src/engine/QueryPlanner.cpp

```diff
@@ -1702,10 +1703,21 @@ QueryPlanner::FiltersAndOptionalSubstitutes QueryPlanner::seedFilterSubstitutes(
   FiltersAndOptionalSubstitutes plans;
   plans.reserve(filters.size());
 
-  // Currently, no filter substitutes are implemented. This will follow in
-  // #1935 and #2140.
-  for (const auto& filterExpression : filters) {
-    plans.emplace_back(filterExpression, std::nullopt);
+  for (const auto& [i, filterExpression] :
+       ::ranges::views::enumerate(filters)) {
+    // Check if the filter expression is suitable for spatial join optimization
+    auto sjConfig = rewriteFilterToSpatialJoinConfig(filterExpression);
+    if (!sjConfig.has_value()) {
+      plans.emplace_back(filterExpression, std::nullopt);
```

Comment:
At some point in the future we probably have to make the substitute aware of the filter it replaces (for the two variable case), but then again we can maybe just choose a different plan.